### DISCUSSION TITLE
Colour/select by entity?

### DISCUSSION THREAD

**Discussion by equinox-graphics** [BradyAJohnston/MolecularNodes#265] [Ideas]
So we have a PDB (8GLV), which is a bit of a monster - it has over a thousand chains, which makes it a bit unmanageable, because we want to colour specific groups of chains different colours.  In comparison, there are ~120 entities, so we could just manually pick out the ones we need and colour them.

Would it be possible to add "entity" to the list of named attributes, or have a node that allows us to select/colour by entity?

We've figured out a workaround (by using R to hack the PDB file and rename all the chains from an entity to a particular name, and then we just use the first 120 chains).  But it's not very elegant!  Alternatively, if there's a way of doing what we want, do let us know.  Have to admit to being new to Blender, so the functionality may well be hiding in there already.  Thanks!

**Comment by BradyAJohnston** [comment]
It certainly is a pretty big structure. I've had a play around with it myself. Currently `entity_id` isn't parsed or added to the structure, but it could certainly be an attribute that could be added.

I'm curious about how you got the entity information. I just had a play around and at least with the `.cif` file  I downloaded from the RCSB, it seems that all atoms have the exact same `entity_id` which is 1.

When I played around with the structure, you can manually select the the vertices of the different types of structures, use `Select Linked` to capture all of the atoms of the chains you want, then create vertex groups based on those manual selections which are then available as attributes inside of geometry nodes. It's a bit janky though, and I think having entity information when it is available would be a good attribute to have.

**Reply by BradyAJohnston** [reply]
Actually I take that back, I was looking at the wrong `.cif` file. Should be able to add `entity_id` as an attribute straight forward enough.

**Reply by equinox-graphics** [reply]
Ah, ok.  That explains it.  No rush to add the feature - we've managed to work around it, but would be super useful for the future, and hopefully for others too!  Thanks!

**Reply by BradyAJohnston** [reply]
Entity ID (colouring / selecting) will be in the next update coming in the next week or so: 

![image](https://github.com/BradyAJohnston/MolecularNodes/assets/36021261/14974db9-3f02-4e84-8ba2-af80f84dd2b2)
